waiter/integration/waiter/metrics_output_test.clj

```diff
@@ -140,3 +140,71 @@
         (is (>= (get-in aggregate-metrics ["counters" "request-counts" "total"]) num-requests)))
 
       (delete-service waiter-url service-id))))
+
+
+(defmacro get-percentile-value
+  [metric p]
+  `(let [p# ~p
+         p-value# (get-in ~metric ["value" p#])]
+     (is (number? p-value#) (str "missing p" p# " value"))
+     p-value#))
+
+(defn- one-started-instance-observed?
+  "Returns true if the launch-metrics state reflects a single started instance
+   in the instance counts for the given service on the given router."
+  [router-url service-id]
+  (-> (make-request router-url "/state/launch-metrics")
+      :body
+      json/read-str
+      (get-in ["state" "service-id->launch-tracker" service-id "instance-counts" "started"])
+      (= 1)))
+
+(deftest ^:parallel ^:integration-slow test-launch-metrics-output
+  (testing-using-waiter-url
+    (let [waiter-settings (waiter-settings waiter-url)
+          metrics-sync-interval-ms (get-in waiter-settings [:metrics-config :metrics-sync-interval-ms])
+          router->endpoint (routers waiter-url)
+          router-urls (vals router->endpoint)
+          service-name (rand-name)
+          sleep-seconds 20
+          min-startup-seconds 10 ; 20s +/- 10s for 2 polls with 5s granularity
+          max-startup-seconds 60 ; the service shouldn't take more than a minute to become healthy
+          instance-count 2
+          req-headers {:x-waiter-cmd (str "sleep " sleep-seconds " && " (kitchen-cmd "-p $PORT0"))
+                       :x-waiter-cmd-type "shell"
+                       :x-waiter-min-instances instance-count
+                       :x-waiter-name service-name}
+          {:keys [headers request-headers service-id] :as first-response}
+          (make-request-with-debug-info req-headers #(make-kitchen-request waiter-url % :method :get))]
+      (with-service-cleanup
+        service-id
+        ; ensure the first request succeded before continuing with testing
+        (assert-response-status first-response 200)
+        ; check that the launch-metrics on each router are present and have sane values
+        (doseq [[router-id router-url] router->endpoint]
+          (wait-for #(one-started-instance-observed? router-url service-id)
+                    :interval 1 :timeout 10)
```

Comment:
Nitpick: does this `10` come from `min-startup-seconds`?

Reply:
Nope. This I copy-pasted the `wait-for` from elsewhere, and decided to keep the same timeout. It's arbitrary. Note that at this point the service has already started (we got the `first-response` back). This `wait-for` is just to ensure that each of the routers has had time to observe the instance's current (running) status.

Reply:
No, but that's a good suggestion! Switched to `min-startup-seconds`.